Add Beats and Elastic Agent feature comparison doc (#2762)
elastic/docs-content: reference/fleet/beats-agent-comparison.md

| {{es}} | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") |
| {{ls}} | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") |
| Kafka | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") |
| Remote {{es}} | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") |
| Redis | ![yes](images/green-check.svg "") | ![no](images/red-x.svg "") | ![no](images/red-x.svg "") |
| File | ![yes](images/green-check.svg "") | ![no](images/red-x.svg "") | ![no](images/red-x.svg "") |
| Console | ![yes](images/green-check.svg "") | ![no](images/red-x.svg "") | ![no](images/red-x.svg "") |

## Supported configurations [supported-configurations]

| {{beats}} configuration | {{agent}} support |
| --- | --- |
| [Modules](beats://reference/filebeat/configuration-filebeat-modules.md) | Supported via integrations. |
| [Input setting overrides](beats://reference/filebeat/advanced-settings.md) | Not configurable. Set to default values. |

> | [General settings](beats://reference/filebeat/configuration-general-options.md) | Many of these global settings are now internal to the agent and for properoperations should not be modified. |

Reviewer: ```suggestion
| [General settings](beats://reference/filebeat/configuration-general-options.md) | Many of these global settings are now internal to the agent and should not be modified. |
```

> | [Project paths](beats://reference/filebeat/configuration-path.md) | {{agent}} configures these paths to provide a simpler and more streamlinedconfiguration experience. |

Reviewer: ```suggestion
| [Project paths](beats://reference/filebeat/configuration-path.md) | {{agent}} configures these paths to provide a simpler and more streamlined configuration experience. |
```

| [Outputs](beats://reference/filebeat/configuring-output.md) | Configured through {{fleet}}. See [Supported outputs](#supported-outputs-beats-and-agent). |
| [SSL](beats://reference/filebeat/configuration-ssl.md) | Supported |
| [{{ilm-cap}}](beats://reference/filebeat/ilm.md) | Enabled by default although the Agent uses [data streams](data-streams.md). |
| [{{es}} index template loading](beats://reference/filebeat/configuration-template.md) | No longer applicable |
| [{{kib}} endpoint](beats://reference/filebeat/setup-kibana-endpoint.md) | New {{agent}} workflow doesn’t need this. |
| [{{kib}} dashboard loading](beats://reference/filebeat/configuration-dashboards.md) | New {{agent}} workflow doesn’t need this. |
| [Processors](beats://reference/filebeat/defining-processors.md) | Processors can be defined at the integration level. Global processors, configured at the policy level, are currently under consideration. |
| [Autodiscover](beats://reference/filebeat/configuration-autodiscover.md) | Autodiscover is facilitated through [dynamic inputs](dynamic-input-configuration.md). {{agent}} does not support hints-based autodiscovery. |

> | [Internal queues](beats://reference/filebeat/configuring-internal-queue.md) | {{fleet}}-managed {{agent}} and Standalone {{agent}} both support configuration of the internal memoryqueues by an end user. Neither support configuration of the internal disk queues by an end user. |

Reviewer: ```suggestion
| [Internal queues](beats://reference/filebeat/configuring-internal-queue.md) | {{fleet}}-managed {{agent}} and Standalone {{agent}} both support configuration of the internal memory queues by an end user. Neither support configuration of the internal disk queues by an end user. |
```

| Data streams support | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | Both {{beats}} (default as of version 8.0) and {{agent}}s use [data streams](data-streams.md) with easier index life cycle management and the [data stream naming scheme](https://www.elastic.co/blog/an-introduction-to-the-elastic-data-stream-naming-scheme). |
| Variables and input conditions | ![no](images/red-x.svg "") | ![yes](images/green-check.svg "") (limited) | ![yes](images/green-check.svg "") | {{agent}} offers [variables and input conditions](dynamic-input-configuration.md) to dynamically adjust based on the local host environment. Users can configure these directly in YAML for standalone {{agent}} or using the {{fleet}} API for {{fleet}}-managed {{agent}}. The Integrations app allows users to enter variables, and we are considering a [UI to edit conditions](https://github.com/elastic/kibana/issues/108525). {{beats}} only offers static configuration. |

> | Allow non-superusers to manage assets and agents | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") (it’s optional) | Starting with version 8.1.0, a superuser role is no longer required to use the {{fleet}} and Integrations apps and corresponding APIs. These apps are optional for standalone {{agent}}. {{beats}} offers [finer grained](beats://reference/filebeat/feature-roles.md) roles. |

Reviewer: ```suggestion
| Allow non-superusers to manage assets and agents | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") (optional) | Starting with version 8.1.0, a superuser role is no longer required to use the {{fleet}} and Integrations apps and corresponding APIs. These apps are optional for standalone {{agent}}. {{beats}} offers [finer grained](beats://reference/filebeat/feature-roles.md) roles. |
```

> | Air-gapped network support | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") (with Limits) | ![yes](images/green-check.svg "") | The {{integrations}} and {{fleet}} apps can be deployed in an air-gapped environment [self-managed deployment of the {{package-registry}}](air-gapped.md#air-gapped-diy-epr). {{fleet}}-managed {{agent}}s require a connection to our artifact repository for agent binary upgrades. However the policy can be modified to have agents point to a local server in order to fetch the agent binary. |

Reviewer: To remain consistent with the other instance where support is limited:

```suggestion
| Air-gapped network support | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") (limited) | ![yes](images/green-check.svg "") | The {{integrations}} and {{fleet}} apps can be deployed in an air-gapped environment [self-managed deployment of the {{package-registry}}](air-gapped.md#air-gapped-diy-epr). {{fleet}}-managed {{agent}}s require a connection to our artifact repository for agent binary upgrades. However the policy can be modified to have agents point to a local server in order to fetch the agent binary. |
```

| Run without root on host | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | {{fleet}}-managed {{agent}}s, Standalone {{agent}}s, and {{beats}} require root permission only if they’re configured to capture data that requires that level of permission. |
| Multiple outputs | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | The policy for a single {{fleet}}-managed {{agent}} can specify multiple outputs. |
| Separate monitoring cluster | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "") | {{fleet}}-managed {{agent}}s, Standalone {{agent}} and {{beats}} can send to a remote monitoring cluster. |

> | Secret management | ![yes](images/green-check.svg "") | ![no](images/red-x.svg "") | ![no](images/red-x.svg "") | {{agent}} stores credentials in the agent policy. We are considering adding [keystore support](https://github.com/elastic/integrations/issues/244). {{beats}} allows users to access credentials in a local [keystore](beats://reference/filebeat/keystore.md). |

Reviewer: > We are considering adding [keystore support](https://github.com/elastic/integrations/issues/244)

The issue was stalled for a long time, and now it's closed. Not sure whether we should keep the link in that case.

> | Progressive or canary deployments | ![yes](images/green-check.svg "") | ![no](images/red-x.svg "") | ![yes](images/green-check.svg "") | {{fleet}} does not have a feature to deploy an {{agent}} policy update progressively but we are considering [improved support](https://github.com/elastic/kibana/issues/108267). With standalone {{agent}} and {{beats}} you can deploy configuration files progressively using third party solutions. |

Reviewer: > but we are considering [improved support](https://github.com/elastic/kibana/issues/108267).

The linked issue was closed as completed and canary deployments for Fleet-managed Elastic Agents are available from 8.19 and 9.1, so we need to revisit this line. Perhaps:

```suggestion
| Progressive or canary deployments | ![yes](images/green-check.svg "") | ![yes](images/green-check.svg "" {applies_to}`stack: ga 9.1.0` | ![yes](images/green-check.svg "") | To upgrade {{fleet}}-managed {{agents}} progressively, you can [configure an automatic upgrade](upgrade-elastic-agent.md#auto-upgrade-agents) in the {{agent}} policy. {applies_to}`stack: ga 9.1.0` </br> With standalone {{agent}} and {{beats}} you can deploy configuration files progressively using third party solutions. |
```
